corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js

```diff
@@ -619,9 +633,29 @@ hqDefine("cloudcare/js/form_entry/form_ui", [
                 }
                 element.pendingAnswer(constants.NO_PENDING_ANSWER);
             } else {
-                response.children = response.tree;
+                const allChildren = response.tree;
                 delete response.tree;
+
+                if (deletedGroup) {
```

Comment:
I think this make sense, but an intro comment would be really helpful to describe what repeat group indexes look like. This logic could also be a good candidate for extracting into a function and then adding tests.

Reply:
Added a comment for now. Will work on pulling out the function tomorrow

Reply:
@orangejenny https://github.com/dimagi/commcare-hq/pull/34467/commits/232019c9ed4af349327f03ff4b6816ba6aeb604c